packages/shared/src/components/squads/moderation/SquadModerationItem.tsx

```diff
@@ -6,71 +6,45 @@ import {
 } from '../../typography/Typography';
 import { CardImage } from '../../cards/common/Card';
 import PostTags from '../../cards/common/PostTags';
-import {
-  SourcePostModeration,
-  SourcePostModerationStatus,
-  verifyPermission,
-} from '../../../graphql/squads';
+import { SourcePostModerationStatus } from '../../../graphql/squads';
 import { SquadModerationActions } from './SquadModerationActions';
 import { ProfileImageSize, ProfilePicture } from '../../ProfilePicture';
 import PostMetadata from '../../cards/common/PostMetadata';
 import { AlertPointerMessage } from '../../alert/common';
-import { SourcePermissions, Squad } from '../../../graphql/sources';
 import { Button, ButtonSize, ButtonVariant } from '../../buttons/Button';
 import { capitalize } from '../../../lib/strings';
-import OptionsButton from '../../buttons/OptionsButton';
 import { TimerIcon, WarningIcon } from '../../icons';
 import { AlertColor } from '../../AlertDot';
-import { useLazyModal } from '../../../hooks/useLazyModal';
-import { LazyModal } from '../../modals/common/types';
 import { useTruncatedSummary } from '../../../hooks';
+import {
+  SquadModerationItemProps,
+  useSourceModerationItem,
+} from './useSourceModerationItem';
+import { SquadModerationItemContextMenu } from './SquadModerationItemContextMenu';
 
-interface SquadModerationListProps {
-  data: SourcePostModeration;
-  onApprove: () => Promise<void>;
-  onReject: () => void;
-  isPending: boolean;
-  squad: Squad;
-}
+export function SquadModerationItem(
+  props: SquadModerationItemProps,
+): ReactElement {
+  const { context, modal, user } = useSourceModerationItem(props);
+  const { data, squad, onApprove, onReject, isPending } = props;
+  const { reason, createdBy, createdAt, image, status } = data;
+
+  const IconComponent =
+    status === SourcePostModerationStatus.Rejected ? WarningIcon : TimerIcon;
 
-export function SquadModerationItem({
-  data,
-  squad,
-  onReject,
-  onApprove,
-  isPending,
-}: SquadModerationListProps): ReactElement {
-  const { status, reason, createdBy, createdAt, image } = data;
   const post = data.sharedPost || data.post;
   const { title } = useTruncatedSummary({
     ...data,
     ...post,
     ...(!post?.title && !data.title && { title: post?.sharedPost?.title }),
   });
-  const isModerator = verifyPermission(squad, SourcePermissions.ModeratePost);
-
-  const { openModal, closeModal } = useLazyModal();
-  const onClick = () => {
-    openModal({
-      type: LazyModal.PostModeration,
-      props: {
-        data,
-        squad,
-        onApprove: () => onApprove().then(closeModal),
-        onReject,
-      },
-    });
-  };
-
-  const IconComponent =
-    status === SourcePostModerationStatus.Rejected ? WarningIcon : TimerIcon;
```

Comment:
Just moved to custom hook because this component were growing too much and naming variable avoiding duplicates was a pain